src/Vehicle/InitialConnectStateMachine.cc

```diff
@@ -198,6 +169,35 @@ void InitialConnectStateMachine::_waitForAutopilotVersionResultHandler(void* res
 
         vehicle->_setCapabilities(autopilotVersion.capabilities);
     }
+        break;
+    case Vehicle::RequestMessageFailureCommandError:
+        qCDebug(InitialConnectStateMachineLog) << QStringLiteral("REQUEST_MESSAGE:AUTOPILOT_VERSION command error(%1)").arg(commandResult);
+        break;
+    case Vehicle::RequestMessageFailureCommandNotAcked:
+        qCDebug(InitialConnectStateMachineLog) << "REQUEST_MESSAGE:AUTOPILOT_VERSION command never acked";
+        break;
+    case Vehicle::RequestMessageFailureMessageNotReceived:
+        qCDebug(InitialConnectStateMachineLog) << "REQUEST_MESSAGE:AUTOPILOT_VERSION command acked but message never received";
+        break;
+    case Vehicle::RequestMessageFailureDuplicateCommand:
+        qCDebug(InitialConnectStateMachineLog) << "REQUEST_MESSAGE:AUTOPILOT_VERSION Internal Error: Duplicate command";
+        break;
+    }
+
+    if (failureCode != Vehicle::RequestMessageNoFailure) {
+        qCDebug(InitialConnectStateMachineLog) << "REQUEST_MESSAGE:AUTOPILOT_VERSION failed. Setting no capabilities";
+        uint64_t assumedCapabilities = 0;
+        if (vehicle->_mavlinkProtocolRequestMaxProtoVersion >= 200) {
+            // Link already running mavlink 2
+            assumedCapabilities |= MAV_PROTOCOL_CAPABILITY_MAVLINK2;
+        }
+        if (vehicle->px4Firmware() || vehicle->apmFirmware()) {
+            // We make some assumptions for known firmware
+            assumedCapabilities |= MAV_PROTOCOL_CAPABILITY_MISSION_INT | MAV_PROTOCOL_CAPABILITY_COMMAND_INT | MAV_PROTOCOL_CAPABILITY_MISSION_FENCE | MAV_PROTOCOL_CAPABILITY_MISSION_RALLY;
```

Comment:
@bkueng This specific change I wonder about. It make some assumptions about known firmwares if the AUTOPILOT_VERSION message is lost. The case where this could theoretically go wrong is when say the vehicle is running modified PX4 firmware which still reports as PX4 but the customization has specifically turned on some of these features. In that case this specific path would turn them back on. What do you think?

Reply:
I think this is ok, as I think it's very rare that someone would turn one of those off. And with the retry logic it should be fairly robust, possibly the most likely failure case is a router in between that consistently drops the message.